salt/_modules/metalk8s_kubernetes.py

```diff
@@ -893,26 +893,48 @@ def delete_service(name, namespace='default', **kwargs):
 
     CLI Examples::
 
-        salt '*' kubernetes.delete_service my-nginx default
-        salt '*' kubernetes.delete_service name=my-nginx namespace=default
+        salt '*' kubernetes.delete_service my-nginx-svc
+        salt '*' kubernetes.delete_service name=my-nginx-svc namespace=default
     '''
     cfg = _setup_conn(**kwargs)
+    body = kubernetes.client.V1DeleteOptions()
 
     try:
-        api_instance = kubernetes.client.CoreV1Api()
-        body = kubernetes.client.V1DeleteOptions()
+        api_instance = kubernetes.client.kubernetes.client.CoreV1Api()
         api_response = api_instance.delete_namespaced_service(
             name=name,
-            body=body,
-            namespace=namespace)
-
-        return api_response.to_dict()
+            namespace=namespace,
+            body=body)
+        mutable_api_response = api_response.to_dict()
+        if not salt.utils.platform.is_windows():
+            try:
+                with _time_limit(POLLING_TIME_LIMIT):
+                    while show_service(name, namespace, **kwargs) is not None:
+                        time.sleep(1)
+                    else:  # pylint: disable=useless-else-on-loop
+                        mutable_api_response['code'] = 200
+            except TimeoutError:
+                pass
+        else:
+            # Windows has not signal.alarm implementation, so we are just falling
+            # back to loop-counting.
+            for i in range(60):
+                if show_service(name, namespace, **kwargs) is None:
+                    mutable_api_response['code'] = 200
+                    break
+                else:
+                    time.sleep(1)
+        if mutable_api_response['code'] != 200:
+            log.warning('Reached polling time limit. Service is not yet '
+                        'deleted, but we are backing off. Sorry, but you\'ll '
```

Comment:
use double quote to avoid escaping the single one (or use `you will` instead, as you prefer).